driver.go

```diff
@@ -257,10 +258,114 @@ func findNeighbors(tx neo4j.Transaction, ids []uint64) ([]neo4jNode, []neo4jNode
 	return sources, targets, edges, nil
 }
 
-func MakeProperties(input map[string]interface{}) map[string]*ls.PropertyValue {
+func MakePropertiesForSetNodeValue(cfg Config, node graph.Node, input map[string]interface{}) interface{} {
+	// check if value type exists in config
+	for k, v := range input {
+		if k != cfg.Map(ls.NodeValueTerm) {
+			continue
+		}
+		vt := cfg.Map(cfg.PropertyTypeMappings[k])
+		if vt != "" {
+			va := ls.GetValueAccessor(vt)
+			switch val := v.(type) {
+			case bool, neo4j.Time, neo4j.LocalTime, neo4j.LocalDateTime, neo4j.Date:
+				if va != nil {
+					form, err := va.FormatNativeValue(val, val, node)
+					if err != nil {
+						panic(fmt.Errorf("cannot format native value for %v, %w", v, err))
+					}
+					return form
+				} else {
+					return val
+				}
+			case float64:
+				if va != nil {
+					form, err := va.FormatNativeValue(val, val, node)
+					if err != nil {
+						panic(fmt.Errorf("cannot format native value for %v, %w", v, err))
+					}
+					return form
+				} else {
+					f := strconv.FormatFloat(v.(float64), 'f', -1, 64)
+					return f
+				}
+			default:
+				return v
+			}
+		} else {
+			switch val := v.(type) {
```

Comment:
Extract out to func: neo4jValueToNativeValue(interface{}) interface{}

Map neo4j.Date/time to types.Date/Time